apps/veil/src/pages/tournament/api/use-personal-rewards.ts

```diff
@@ -1,30 +1,105 @@
 import { useQuery } from '@tanstack/react-query';
 import { connectionStore } from '@/shared/model/connection';
-import { penumbra } from '@/shared/const/penumbra';
-import { ViewService } from '@penumbra-zone/protobuf';
 import { AddressIndex } from '@penumbra-zone/protobuf/penumbra/core/keys/v1/keys_pb';
+import { ViewService } from '@penumbra-zone/protobuf';
+import { penumbra } from '@/shared/const/penumbra';
+import { statusStore } from '@/shared/model/status';
+import { TournamentDelegatorHistoryResponse } from '../../../shared/api/server/tournament/delegator-history';
+import { apiPostFetch } from '@/shared/utils/api-fetch';
+import { TournamentVotesResponse } from '@penumbra-zone/protobuf/penumbra/view/v1/view_pb';
+
+const fetchRewards = async (
+  subaccount?: number,
+  epochOrHeight?: { type: 'epoch'; value: bigint } | { type: 'blockHeight'; value: bigint },
+): Promise<TournamentDelegatorHistoryResponse> => {
+  let votes: TournamentVotesResponse[];
+  const accountFilter =
+    typeof subaccount === 'undefined' ? undefined : new AddressIndex({ account: subaccount });
+
+  // `tournamentVotes` accepts either `blockHeight` or `epochIndex`:
+  //  * `blockHeight` – calls `iterateLQTVotes` to gather every vote up to the
+  //   epoch containing that height, grouping results by epoch.
+  //  * `epochIndex` – returns the votes for that single epoch, already grouped.
+
+  if (epochOrHeight?.type === 'epoch') {
+    votes = await Array.fromAsync(
+      penumbra
+        .service(ViewService)
+        .tournamentVotes({ accountFilter, epochIndex: epochOrHeight.value }),
+    );
+  } else if (epochOrHeight?.type === 'blockHeight') {
+    votes = await Array.fromAsync(
+      penumbra.service(ViewService).tournamentVotes({
+        accountFilter,
+        blockHeight: epochOrHeight.value,
+      }),
+    );
+  }
+
+  const { address } = await penumbra
+    .service(ViewService)
+    .addressByIndex({ addressIndex: { account: accountFilter?.account } });
```

Comment:
suggestion: move this async request into a promise.all along with the previous .tournamentVotes call